source/Calamari.Aws/Integration/AwsScriptWrapper.cs

```diff
@@ -16,7 +19,11 @@ public class AwsScriptWrapper : IScriptWrapper
         readonly ILog log;
         readonly IVariables variables;
         public int Priority => ScriptWrapperPriorities.CloudAuthenticationPriority;
-        bool IScriptWrapper.IsEnabled(ScriptSyntax syntax) => true;
+
+        bool IScriptWrapper.IsEnabled(ScriptSyntax syntax) =>
+            variables.Get(SpecialVariables.Account.AccountType) == "AmazonWebServicesAccount" ||
+            string.Equals(variables.Get(AwsSpecialVariables.Authentication.UseInstanceRole), bool.TrueString,
+                StringComparison.InvariantCultureIgnoreCase);
```

Comment:
Felt a bit clunky to read this, might be worth modifying it to something like this:
```suggestion
        bool IScriptWrapper.IsEnabled(ScriptSyntax syntax)
        {
            var accountType = variables.Get(SpecialVariables.Account.AccountType);
            var useAwsInstanceRole = variables.Get(AwsSpecialVariables.Authentication.UseInstanceRole);
            return accountType == "AmazonWebServicesAccount" ||
                string.Equals(useAwsInstanceRole, bool.TrueString, StringComparison.InvariantCultureIgnoreCase);
         }
```